File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        output=ListNode(0)
        current=output
        carry=0
        while l1!=None or l2!=None or carry!=0:
            l1val=l1.val if l1 else 0
            l2val=l2.val if l2 else 0
            sum=l1val+l2val+carry
            carry=sum//10
            current.next = ListNode(sum % 10)
            current = current.next        
            l1=l1.next if l1 else None
            l2=l2.next if l2 else None
        return output.next

File: 0002-add-two-numbers/0002-add-two-numbers.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function addTwoNumbers(l1: ListNode | null, l2: ListNode | null): ListNode | null {
    let output = new ListNode(0);
    let currentNode = output;
    let carry = 0;
    while (l1 || l2) {
        let sum = (l1 ? l1.val : 0) + (l2 ? l2.val : 0) + carry;
        carry = sum > 9 ? 1 : 0;
        currentNode.next = new ListNode(sum%10);
        currentNode = currentNode.next;
        l1 = l1 ? l1.next : null;
        l2 = l2 ? l2.next : null;
    }
    if (carry) currentNode.next = new ListNode(carry)
    return output.next;
};